Usdz update

GitHubDragonFly opened the pull request:
Related issue: #26616

**Description**

Separate exported material properties entries so they are set individually and unconditionally.
Add new MeshPhysicalMaterial properties: specularColor / specularColorMap / clearcoatMap / clearcoatRoughnessMap

GitHubDragonFly committed:
Update `buildMaterial` function in USDZ Exporter

- Separate current material properties (inputs/sampler) entries
- Add new MeshPhysicalMaterial entries: specularColor / specularColorMap / clearcoatMap / clearcoatRoughnessMap

GitHubDragonFly committed:
Restore `defaultPrim` entry

GitHubDragonFly committed:
Update USDZ Loader

- Add `usda` filename check
- Add data check to the `findGeometry` function
- Add missing string `Mesh` to `def` constant in the `findGeometry` function
- Add moving `texCoord2f[] primvars:st` to Mesh
- Replace MeshStandardMaterial with MeshPhysicalMaterial to cover all material properties
- Update `buildMaterial` function by separating current material properties entries and adding new MeshPhysicalMaterial entries: specularColor / specularColorMap / clearcoatMap / clearcoatRoughnessMap
- Add map colorSpace entries

Mugen87 commented:
I was expecting to see a much smaller PR with the changes mentioned in https://github.com/mrdoob/three.js/issues/26616#issuecomment-1685379989. Could we do that first?

Large PRs are in general hard to review so it's best to break things down into smaller one.

GitHubDragonFly commented:
All the changes in this PR go hand in hand with each other.

Smaller changes that were mentioned still required moving texture coordinates code to the Mesh so the main texture map could be seen.

All other changes are only related to material properties, which when changed in the Exporter should be changed/reflected in the Loader as well.

Feel free to reject this PR. It can be used in the future as a code reference for anyone who might be looking to do smaller changes.

GitHubDragonFly committed:
Update USDZ Exporter

- Remove pure null checks to allow for undefined as well
- Add some material property existence dependent processing

Mugen87 commented:
Closing. We really need to split up things into smaller PRs.
